src/domain_bridge/domain_bridge.cpp

```diff
@@ -234,6 +320,17 @@ class DomainBridgeImpl
     return publisher;
   }
 
+  /// Check if there is a pulisher with the given gid
+  bool has_publisher_with_gid(const rmw_gid_t & _publisher_gid)
+  {
+    for (auto it = bridged_topics_.begin(); it != bridged_topics_.end(); ++it) {
+      if (*it->second.first->get_base_publisher() == _publisher_gid) {
```

Comment:
Thanks for looking into this, we were having the same issue and this PR helped us. I noticed one issue when we tried using this; we got a segmentation fault on this line. I haven't looked into the root cause, but adding this patch fixed the issue.

```suggestion
      if (it->second.first && *it->second.first->get_base_publisher() == _publisher_gid) {
```